jobqueue.go

```diff
@@ -110,11 +149,35 @@ func (s *Stack) run() {
 				j := s.stack.pop()
 				j.notify <- nil
 			}
+
+			if s.closing && s.busy == 0 && s.stack.empty() {
+				close(s.hasQuit)
+				return
+			}
 		case <-timeout:
 			oldest.notify <- ErrTimeout
 			s.stack.shift()
-		case <-s.quit:
-			// TODO: teardown
+		case status := <-s.status:
+			status <- Status{ActiveJobs: s.busy, QueuedJobs: s.stack.list.Len(), Closing: s.closing}
```

Comment:
Can this block the whole loop or are we sure that the chan of chan is always empty to write to?

Reply:
It won't block, and yes we can be sure, because the channel is created and read from, by internal code.